### DISCUSSION TITLE
Figuring out which IR protocol a mystery device uses

### DISCUSSION THREAD

**Discussion by guillochon** [micropython/micropython#9340] [Libraries & Drivers]
Hi all, I have a range hood that is missing its remote control. I've learned serendipitously that the range hood can be controlled by an electronic candle remote I had laying around. I wanted to build a Pico project where I can control the range hood with an IR transmitter, and armed with this candle remote I was hoping to use the `micropython_ir` library and an IR receiver to figure out which codes are being transmitted so I can reproduce them.

My question is how do I know which protocol this remote uses? I assume it must be a relatively common one, but is the best way trial and error with each of the protocols? @peterhinch

**Comment by jimmo** [comment]
> I've learned serendipitously that the range hood can be controlled by an electronic candle remote I had laying around.
> My question is how do I know which protocol this remote uses?

What equipment do you have? An oscilloscope or logic analyser?

**Comment by guillochon** [comment]
@jimmo Nothing other than what is described in the `micropython_ir` package (an IR receiver and some basic electronics). I haven't received the hardware yet but I figured I'd pose the question now in case I needed to get something else.

**Comment by fdufnews** [comment]
In the [package](https://github.com/peterhinch/micropython_ir), there is an ir_rx example which returns the name of the protocol if it is recognized by the lib.
You can first give it a try and see if it is a known one.

**Reply by peterhinch** [reply]
Yes, it is documented [here](https://github.com/peterhinch/micropython_ir/blob/master/RECEIVER.md#21-test-scripts).
